python/pyspark/sql/types.py

```diff
@@ -1249,7 +1249,7 @@ def _infer_schema_type(obj, dataType):
 }
 
 
-def _verify_type(obj, dataType, nullable=True):
+def _verify_type(obj, dataType, nullable=True, name="obj"):
```

Comment:
I meant this case:

```python
>>> from pyspark.sql.types import *
>>> spark.createDataFrame(["a"], StringType()).printSchema()
```

```
root
 |-- value: string (nullable = true)
```
```python
>>> from pyspark.sql.types import *
>>> spark.createDataFrame(["a"], IntegerType()).printSchema()
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../spark/python/pyspark/sql/session.py", line 526, in createDataFrame
    rdd, schema = self._createFromLocal(map(prepare, data), schema)
  File ".../spark/python/pyspark/sql/session.py", line 387, in _createFromLocal
    data = list(data)
  File ".../spark/python/pyspark/sql/session.py", line 516, in prepare
    verify_func(obj, dataType)
  File ".../spark/python/pyspark/sql/types.py", line 1326, in _verify_type
    % (name, dataType, obj, type(obj)))
TypeError: obj: IntegerType can not accept object 'a' in type <type 'str'>
```

It sounds "obj" should be "value". It looks we should specify the name around https://github.com/dgingrich/spark/blob/topic-spark-19507-verify-types/python/pyspark/sql/session.py#L516.